scripts/bert/finetune_classifier.py

```diff
@@ -424,15 +431,61 @@ def train(metric):
         model.save_parameters(params_saved)
         logging.info('params saved in : {0}'.format(params_saved))
         toc = time.time()
-        logging.info('Time cost=%.1fs', toc - tic)
+        logging.info('Time cost=%.2fs', toc - tic)
         tic = toc
 
 
+def inference(metric):
+    """Inference function."""
+
+    logging.info('Now we are doing BERT classification inference on {} !'.format(ctx))
+    model = BERTClassifier(bert, dropout=0.1, num_classes=len(task.get_labels()))
+    model.hybridize()
+    model.load_parameters(model_parameters, ctx=ctx)
+
+    metric.reset()
+    step_loss = 0
+    tic = time.time()
+    for batch_id, seqs in enumerate(dev_data):
+        input_ids, valid_length, type_ids, label = seqs
+        out = model(input_ids.as_in_context(ctx),
+                    type_ids.as_in_context(ctx),
+                    valid_length.astype('float32').as_in_context(ctx))
+
+        ls = loss_function(out, label.as_in_context(ctx)).mean()
+
+        step_loss += ls.asscalar()
+        metric.update([label], [out])
+
+        if (batch_id + 1) % (args.log_interval) == 0:
+            metric_nm, metric_val = metric.get()
+            if not isinstance(metric_nm, list):
+                metric_nm = [metric_nm]
+                metric_val = [metric_val]
+            eval_str = '[Batch {}/{}] loss={:.4f}, metrics=' + \
+                ','.join([i + ':{:.4f}' for i in metric_nm])
+            logging.info(eval_str.format(batch_id + 1, len(dev_data), \
+                step_loss / args.log_interval, \
+                *metric_val))
+            step_loss = 0
+
+    mx.nd.waitall()
+    toc = time.time()
+    total_num = dev_batch_size * len(dev_data)
+    logging.info('Time cost={:.2f} s throughput={:.2f} samples/s'.format(
+        toc - tic, total_num / (toc - tic)))
```

Comment:
lazy logging
```python
logging.info('Time cost=%.2f s throughput=%.2f samples/s',  toc - tic, total_num / (toc - tic))
```
Other logs should also consider lazy mode.

Reply:
Thank you for your review @kenjewu. Why do you think lazy logging is better here?

Reply:
If use lazy construction of the string, the logging string is not actually constructed unless logging actually happens. This will be more efficient

Reply:
Thanks @kenjewu , and doubt that does the code below need to do lazy logging?
`logging.info('params saved in : {0}'.format(params_saved))`

Reply:
not really, but this would: `logging.info('params saved in : {0}', params_saved)`

Reply:
Thank you @kenjewu, @szha, @pengxin99 . Now it's fixed.